### PR TITLE
add positive ut for service account projection

### PR DESCRIPTION
**Pull request by krmayankk**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
adds a UT

**Which issue(s) this PR fixes**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/core/validation/validation_test.go [hunk 1]
```diff
@@ -6686,6 +6686,33 @@ func TestValidatePod(t *testing.T) {
 				DNSPolicy:     core.DNSClusterFirst,
 			},
 		},
+		{ // valid serviceaccount token projected volume with serviceaccount name specified
+			ObjectMeta: metav1.ObjectMeta{Name: "valid-extended", Namespace: "ns"},
```
Comment (mourya007) [inline pkg/apis/core/validation/validation_test.go]: Just a query that , Why we are not specifying expectedError: here.?

Comment (mikedanese) [reply pkg/apis/core/validation/validation_test.go]: Because we don't expect an error.

Comment (liggitt) [reply pkg/apis/core/validation/validation_test.go]: no error is expected

**Pull request merged**